types/Page.php

```diff
@@ -78,25 +100,26 @@ public function handleAjax()
         $max = $this->config['autocomplete']['maxresult'];
         if ($max <= 0) return [];
 
-        // lookup with namespace and postfix applied
-        $namespace = $this->config['autocomplete']['namespace'];
-        if ($namespace) {
-            // namespace may be relative, resolve in current context
-            $namespace .= ':foo'; // resolve expects pageID
-            $resolver = new PageResolver($INPUT->str('ns') . ':foo'); // resolve relative to current namespace
-            $namespace = $resolver->resolveId($namespace);
-            $namespace = getNS($namespace);
-        }
-        $postfix = $this->config['autocomplete']['postfix'];
-        if ($namespace) $lookup .= ' @' . $namespace;
-
         $data = ft_pageLookup($lookup, true, $this->config['usetitles']);
         if ($data === []) return [];
 
-        // this basically duplicates what we do in ajax_qsearch()
+        $filter = $this->config['autocomplete']['filter'];
+
+        // try to use deprecated options namespace and postfix as filter
+        $namespace = $this->config['autocomplete']['namespace'] ?? '';
+        $postfix = $this->config['autocomplete']['postfix'] ?? '';
+        if (!$filter) {
+            $filter = $namespace ? $namespace . ':' : '';
+            $filter .= $postfix ? '.+?' . $postfix . '$' : '';
+        }
```

Comment:
Is repeating this here necessary? Isn't `getConfig()` executed when the object is instantiated anyway?

Reply:
Repeating the code was necessary because I applied the migration in the wrong place to begin with. I moved it from `getConfig()`  to `mergeConfig()`, which is called when instantiating the object.

The redundant code is removed.